File: server/Models/Vault.cs
namespace keeprFINAL.Models;

public class Vault : RepoItem<int>
{
	// NOTE: RepoItem fills in the Id
	// public string Id { get; set; }

	public string CreatorId { get; set; }
	public string Name { get; set; }
	public string Description { get; set; }
	public string Img { get; set; }
	public bool IsPrivate { get; set; }


	public Profile Creator { get; set; }

}

File: client/src/models/Vault.js
import { DatabaseItem } from "./DatabaseItem.js";
import { Profile } from "./Profile.js";

export class Vault extends DatabaseItem {
   constructor(data) {
      super(data) //calls constructor on DataBaseItem

    // id, createdAt, updatedAt inherited from DatabaseItem

		this.creatorId = data.creatorId
		this.name = data.name
		this.description = data.description
		this.img = data.img
		this.isPrivate = data.isPrivate

		
		// NOTE: one will just be an object, and one will be a classed object, but they could end up working the exact same, depending on how your backend is set up. The first one is the preferred method though, cause it will type the creator property, which gives you intellisense on that property in vscode. 
		this.creator = new Profile(data.creator)
		// this.creator = data.creator
  }
}
